product/src/controllers/userController.js

```diff
@@ -0,0 +1,41 @@
+import express from 'express';
+import bcrypt from 'bcryptjs';
+import jwt from 'jsonwebtoken';
+import { Users } from '../models/users.js';
+
+export class UserController {
+   async registerAccount(req, res) {
+        const user = req.body;
+        user.password = await bcrypt.hash(user.password, 10);
+
+        await Users.create(user)
+            .then(() => {
+                res.status(201).json(user);
+            })
+            .catch((error) => {
+                res.status(400).json({ message: error.message });
+            })
+    };
+    
+   async generateToken(req, res) {
+        const user = await Users.findOne({ 
+            attributes: ['id', 'name', 'email', 'password'],
+            where: {
+                email: req.body.email
+             }
+            });
+                
+        if(!user)
+            return res.status(400).send({ error: 'User not found.' });
+    
+        if(!(await bcrypt.compare(req.body.password, user.password)))
```

Comment:
i think the right approach is to create a variable with the result before and put the variable into if